core/api/serializers/project_history.py

```diff
@@ -8,49 +8,26 @@
 User = get_user_model()
 
 
-class ProjectHistorySerializer(serializers.ModelSerializer):
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ["id", "username", "email", "first_name", "last_name"]
 
+
+class ProjectHistorySerializer(serializers.ModelSerializer):
     project_id = serializers.PrimaryKeyRelatedField(
         required=False,
         queryset=Project.objects.all().values_list("id", flat=True),
         write_only=True,
     )
 
-    updated_by_id = serializers.PrimaryKeyRelatedField(
-        required=False,
-        queryset=User.objects.all().values_list("id", flat=True),
-        write_only=True,
-    )
-
-    updated_by_username = serializers.StringRelatedField(
-        read_only=True,
-        source="updated_by.username",
-    )
-
-    updated_by_email = serializers.StringRelatedField(
-        read_only=True,
-        source="updated_by.email",
-    )
-
-    updated_by_first_name = serializers.StringRelatedField(
-        read_only=True,
-        source="updated_by.first_name",
-    )
-
-    updated_by_last_name = serializers.StringRelatedField(
-        read_only=True,
-        source="updated_by.last_name",
-    )
+    user = UserSerializer(read_only=True)
```

Comment:
Does UserSerializer work with read_only=True? I expected it to not work on `history_serializer.save(user=request_user)`

Reply:
It does work, I think it would fail if added to the data dictionary: `ProjectHistorySerializer(data=history_data)`. But it works when passed to `.save`.